src/export/csv/index.ts

```diff
@@ -68,13 +71,21 @@ export function computeCSVData({
     },
   ];
 
-  // check if the passport has payment or submission ids, add them as reference rows if exist
   const passportInstance = new Passport(passport);
-  const conditionalKeys = [
-    "application.fee.reference.govPay",
-    "bopsId",
-    "idoxSubmissionId",
-  ];
+
+  // check if the passport has payment info (or session govUkPayment if invite to pay), add as reference row if exists
+  const selfPayPassportKey = "application.fee.reference.govPay";
+  if (passportInstance.has([selfPayPassportKey]) || govUkPayment) {
+    references.push({
+      question: selfPayPassportKey,
+      responses:
+        (passportInstance.any([selfPayPassportKey]) as Response[]) ||
+        (govUkPayment as unknown as Response[]),
```

Comment:
I know this is ugly type casting, but does what we need with minimal cascading changes to these bits of old tedious code !